controller/konnect/konnectextension_controller.go

```diff
@@ -326,82 +562,45 @@ func (r *KonnectExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
-			Namespace: ext.Namespace,
-			Name:      ext.Spec.KonnectConfiguration.APIAuthConfigurationRef.Name,
-		}, nil
+	if updateExtensionStatus {
+		log.Debug(logger, "updating KonnectExtension status")
+		return ctrl.Result{}, r.Client.Status().Update(ctx, &ext)
 	}
 
-	// In case the KonnectConfiguration is not set, we fetch the KonnectGatewayControlPlane
-	// and get the KonnectConfiguration from `spec.konnectControlPlane.controlPlane.konnectNamespacedRef`.
-	// KonnectGatewayControlPlane reference and KonnectConfiguration
-	// are mutually exclusive in the KonnectExtension API.
-	cpRef := ext.Spec.KonnectControlPlane.ControlPlaneRef.KonnectNamespacedRef
-	kgcp := &konnectv1alpha1.KonnectGatewayControlPlane{}
-	err := cl.Get(ctx, client.ObjectKey{
-		// TODO: handle cross namespace refs to KonnectGatewayControlPlane when referencing CP from another namespace is supported.
-		Namespace: ext.Namespace,
-		Name:      cpRef.Name,
-	}, kgcp)
-	if err != nil {
-		return types.NamespacedName{}, err
+	readyCondition = metav1.Condition{
+		Type:    konnectv1alpha1.KonnectExtensionReadyConditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  konnectv1alpha1.KonnectExtensionReadyReasonReady,
+		Message: "KonnectExtension is ready",
 	}
-	return types.NamespacedName{
-		Namespace: kgcp.Namespace,
-		Name:      kgcp.Spec.KonnectConfiguration.APIAuthConfigurationRef.Name,
-	}, nil
-}
 
-func getCertificateSecretRef(ctx context.Context, cl client.Client, ext konnectv1alpha1.KonnectExtension) (types.NamespacedName, error) {
-	var certificateSecret corev1.Secret
-	switch *ext.Spec.DataPlaneClientAuth.CertificateSecret.Provisioning {
-	case konnectv1alpha1.ManualSecretProvisioning:
-		// No need to check CertificateSecretRef is nil, as it is enforced at the CRD level.
-		if err := cl.Get(ctx, types.NamespacedName{
-			Namespace: ext.Namespace,
-			Name:      ext.Spec.DataPlaneClientAuth.CertificateSecret.CertificateSecretRef.Name,
-		}, &certificateSecret); err != nil {
-			return types.NamespacedName{}, err
-		}
-	default:
-		return types.NamespacedName{}, errors.New("automatic secret provisioning not supported yet")
+	if res, updated, err := patch.StatusWithConditions(
+		ctx,
+		r.Client,
+		&ext,
+		readyCondition,
+	); err != nil || updated || !res.IsZero() {
+		return res, err
 	}
-	return client.ObjectKeyFromObject(&certificateSecret), nil
-}
 
-func konnectClusterTypeToCRDClusterType(clusterType sdkkonnectcomp.ControlPlaneClusterType) konnectv1alpha1.KonnectExtensionClusterType {
-	switch clusterType {
-	case sdkkonnectcomp.ControlPlaneClusterTypeClusterTypeControlPlane:
-		return konnectv1alpha1.ClusterTypeControlPlane
-	case sdkkonnectcomp.ControlPlaneClusterTypeClusterTypeK8SIngressController:
-		return konnectv1alpha1.ClusterTypeK8sIngressController
-	default:
-		// default never happens as the validation is at the CRD level
-		return ""
-	}
+	log.Debug(logger, "reconciled")
+	return ctrl.Result{
+		RequeueAfter: r.syncPeriod,
+	}, nil
```

Comment:
Nit: I think we could add a comment here (similar to the one in generic reconciler) stating this is is to enforce the config in Konnect. WDYT?